autonav_ws/src/autonav_odometry/src/main.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+# filters.py from last year for the most part
+
+import json
+from types import SimpleNamespace
+from autonav_msgs.msg import MotorFeedback, GPSFeedback, Position
+from autonav_shared.types import DeviceState, SystemState, LogLevel # i am goig to beat pylance with a rock
+from particlefilter import ParticleFilter
+# from scr_msgs.msg import SystemState
+from autonav_shared.node import Node
+from enum import IntEnum
+import rclpy
+
+class FilterType(IntEnum):
+    PARTICLE_FILTER = 1
+    # ADD NEW FILTERS HERE
+
+class FiltersNodeConfig:
+    def __init__(self):
+        self.filter_type = FilterType.PARTICLE_FILTER
+
+class FiltersNode(Node):
+    def __init__(self):
+        super().__init__("autonav_odometry")
+        # init gps and position
+        self.first_gps = None
+        self.last_gps = None
+        self.latitude_length = self.declare_parameter("latitude_length", 111086.2).get_parameter_value().double_value
+        self.longitude_length = self.declare_parameter("longitude_length", 81978.2).get_parameter_value().double_value
```

Comment:
this might be more of an in-person conversation (maybe probably with Tony) but I think lat and long length should be part of the config instead of being launch file parameters.